Web UI (topology graph): Show FQDN for nodes if they have no common DNS zone

serg-cymbaluk opened the pull request:
It allows to avoid confusion with identical short hostnames.

There are two cases implemented:
- no common DNS zone: graph shows FQDN for all nodes
- all nodes have one common DNS zone: graph shows DN relatively to the common zone

https://pagure.io/freeipa/issue/7206

flo-renaud commented on install/ui/test/topology_tests.js:
```diff
@@ -0,0 +1,147 @@
+    var nodes = [
+        { id: 'master.zone1' },
+        { id: 'replica.zone1' },
+        { id: 'master.zone2' },
```
FreeIPA does not allow single label domains, you could use instead master.sub.zone1 replica.sub.zone1, master.sub.zone2.

serg-cymbaluk replied:
Fixed

serg-cymbaluk commented:
@flo-renaud 
I've moved "common domain" logic from TopoGraph to TopologyGraphWidget and so it allows me to add this domain name to the page title. Tests are also updated.

flo-renaud commented:
Hi @serg-cymbaluk 
I noticed the following behavior. If I install 2 nodes with the same domain, the graph properly shows "Topology graph (domain.com)" and only the short names are displayed below the nodes. Then I add a 3rd node with a completely different domain replica.example.test and hit the "refresh" button. The "Topology graph (domain.com)" label does not get updated, whereas the FQDN are displayed for the nodes.
If I reload the page with Ctrl-R, the topology graph label gets updated ("Topology graph" without any domain name).
So it seems that refresh button does not refresh the "Topology graph" label.

serg-cymbaluk commented:
@flo-renaud 
> The "Topology graph (domain.com)" label does not get updated, whereas the FQDN are displayed for the nodes.

This behavior has been fixed.

flo-renaud commented:
Hi @serg-cymbaluk 
the fix works for me, thanks. ACK

serg-cymbaluk commented:
master:

* 133b199f61a6756723a9a5869f7ff7e415bea455 Web UI (topology graph): Show FQDN for nodes if they have no common DNS zone
